lib/firewall/firewall.cc

```diff
@@ -786,9 +853,17 @@ bool ethernet_driver_start(std::atomic<uint8_t> *state)
 	}
 	Debug::log("Initialising network interface");
 	auto &ethernet = lazy_network_interface();
-	ethernet.mac_address_set();
+	ethernet.mac_address_set(mac_address());
 	// Poke the barrier and make the driver thread start.
 	barrier = 2;
 	barrier.notify_one();
 	return true;
 }
+
+uint8_t *firewall_mac_address_get()
+{
+	CHERI::Capability ret{mac_address().data()};
+	ret.permissions() &= {CHERI::Permission::Load, CHERI::Permission::Global};
+	ret.bounds() = 6;
```

Comment:
Just for my own edification, what are the bounds absent this line?

Reply:
Looks like it's 6.  I'll change this to an assertion.